compiler/rustc_lint/src/non_fmt_panic.rs

```diff
@@ -61,6 +63,25 @@ impl<'tcx> LateLintPass<'tcx> for NonPanicFmt {
                             check_panic(cx, f, arg);
                         }
                     }
+                } else if matches!(f_diagnostic_name, Some(sym::unreachable_display)) {
+                    if let Some(id) = f.span.ctxt().outer_expn_data().macro_def_id {
+                        if matches!(
+                            cx.tcx.get_diagnostic_name(id),
+                            Some(sym::unreachable_2015_macro)
+                        ) {
```

Comment:
If you're only checking a single diagnostic name + def_id pair, you can use `tcx.is_diagnostic_item(sym, id)` instead.
```suggestion
                        if cx.tcx.is_diagnostic_item(sym::unreachable_2015_macro, id) {
```